src/main/java/io/nats/client/impl/NatsFeatureBase.java

```diff
@@ -71,6 +71,7 @@ protected MessageInfo _getBySeq(long seq) throws IOException, JetStreamApiExcept
 
     protected void visitSubject(String subject, DeliverPolicy deliverPolicy, boolean headersOnly, boolean ordered, MessageHandler handler) throws IOException, JetStreamApiException, InterruptedException {
         PushSubscribeOptions pso = PushSubscribeOptions.builder()
+            .stream(streamName)
```

Comment:
It's always better to supply the stream when you know it, this should have been there originally.